packages/workflow/src/features/correction/routes.ts

```diff
@@ -293,28 +298,56 @@ export const routes = [
       const { correction, ...registration } =
         recordInputWithUploadedAttachments.registration!
 
+      /*
+       * Separate tasks from the record
+       */
+      const [correctionRequestApprovedTask, correctedTask] =
+        getTasksInAscendingOrder(recordInCorrectedState)
+      const recordWithoutTasks = getRecordWithoutTasks(recordInCorrectedState)
+
+      /*
+       * Create and update the valid record that only has one task (the corrected task)
+       */
+      const newRecord = addTaskToRecord(recordWithoutTasks, correctedTask)
+
+      newRecord satisfies ValidRecord
+
       const recordWithUpdatedValues = updateFHIRBundle(
-        recordInCorrectedState,
+        newRecord,
```

Comment:
Now at this point the record only has the "corrected" task that we want to have as the final state of the record